Improve ping level documentation (#550)
mozilla/firefox-data-docs: src/concepts/choosing_a_dataset.md

If you want to know whether a causal link occurs between two events, you can learn more at [tools for experimentation](../tools/experiments.md).

There are two types of datasets that you might want to use: those based on raw pings and those derived from them.

## Raw Ping Datasets

We receive data from Firefox users via **pings**: small JSON payloads sent by clients at specified intervals.
There are many types of pings, each containing different measurements and sent for different purposes.

These pings are then [aggregated into ping-level datasets](../cookbooks/bigquery/querying.md#structure-of-ping-tables-in-bigquery) that can be retrieved using BigQuery.
Pings can be a bit difficult to work with in their raw form and there are many pitfalls that are easy to fall into when using them naively.

> Where possible, you should use a derived dataset to answer your question.

Reviewer: Do we still believe this? `main_summary` is described as deprecated; _that_ page tells you to use `telemetry.main`. If we don't have a practice of adding new columns to `clients_daily` and `clients_last_seen` then there are classes of analysis they won't support.

I also think the "raw ping" phraseology is a throwback to a time where the storage format and access semantics were meaningfully different from the derived datasets; maybe this is out of scope for this change but it probably makes sense to just call them "ping tables" now.

Author: I'd probably describe `main_summary` less as a derived dataset and more as a precursor to the ping tables. Other than that, I agree with you that the wording could be better, let's try to do that while we're here.

> These pings are then [aggregated into ping-level datasets](../cookbooks/bigquery/querying.md#structure-of-ping-tables-in-bigquery) that can be retrieved using BigQuery.
> Pings can be a bit difficult to work with in their raw form and there are many pitfalls that are easy to fall into when using them naively.
> Where possible, you should use a derived dataset to answer your question.

Author: How about this?

```suggestion
These pings are then [aggregated into ping-level datasets](../cookbooks/bigquery/querying.md#structure-of-ping-tables-in-bigquery) that can be retrieved using BigQuery.
Pings can be difficult to work with and expensive to query: where possible, you should use a derived dataset to answer your question.
```

Reviewer: sgtm!

For more information on pings and how to use them, see [Raw Ping Data](../datasets/pings.md).

## Derived Datasets

> Derived datasets are built using the raw ping data above with various transformations to make them easier to work with and help you avoid common pitfalls.

Reviewer: I guess I wonder which pitfalls derived datasets generally help you avoid. clients_daily in particular lends itself to _different_ pitfalls (once-per-session metrics are often misaggregated).

Author: I'm not sure what the original intent was, but I was thinking mainly of pseudo-replication. How about this?

```suggestion
Derived datasets are built using the raw ping data above with various transformations to make them easier to work with and help you avoid the pitfall of [pseudo-replication](https://docs.telemetry.mozilla.org/concepts/analysis_gotchas.html#pseudo-replication).
```

Reviewer: sgtm!
